src/main/java/spring/boot/app/demo/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,29 @@
+package spring.boot.app.demo.service.impl;
+
+import javax.persistence.EntityNotFoundException;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import spring.boot.app.demo.model.User;
+import spring.boot.app.demo.service.UserService;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String userId) throws UsernameNotFoundException {
+        User userFromDB = userService.findByNativeId(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can not find user with id: " + userId));
```

Comment:
```suggestion
                .orElseThrow(() -> new UsernameNotFoundException("Can not find user with id: " + userId));
```
думаю тут логічніше буде кинули UsernameNotFoundException так як це типова поведінка для loadUserByUsername ( в сигнатурі методу присутній цей ексепшин)